DTS-STN/passport-status
# Added cancel button with modal dialog

## [ADO-989](https://dev.azure.com/DTS-STN/passport-status/_workitems/edit/989)

### Description
This PR adds the `CancelModal` component which renders an `ActionButton` where, upon click, the user is presented with a modal dialog. I've made the component re-usable in the sense that we can supply the modal dialog as many buttons as we want (so perhaps renaming the component to just `Modal` would be better? Open to thoughts on this). For our use case, when a user clicks the cancel button they are asked `Are you sure you want to go back?`. If they answer `Yes`, they will be redirected to the landing page. If they select `No`, they dialog will close and they can continue to fill out the form.

This also brings in the `react-focus-on` package, which locks focus within the modal, removes scrolling functionality (and hides the scrollbar) when focus is within the modal, and also applies aria-hidden to the rest of the page when within the modal to prevent screen readers from announcing anything on the page aside from the modal content.

### What to test for/How to test
1. Pull in branch
2. Type `npm run dev`
3. Navigate to `/status`
4. Click Cancel
5. Clicking `Yes` should redirect you back to `/landing`
6. Clicking `No` should close the modal
